gdrdrv/gdrdrv.c

```diff
@@ -479,16 +493,23 @@ static int gdrdrv_pin_buffer(gdr_info_t *info, void __user *_params)
         }
     }
 
-
     // here a typical driver would use the page_table to fill in some HW
     // DMA data structure
 
-    params.handle = mr->handle;
-
     mutex_lock(&info->lock);
     list_add(&mr->node, &info->mr_list);
+
+    // The user treats handle as the offset value for mmap. To guarantee no
+    // overlapping, the next offset will sit next to this mapping range.
+    // Remember that the address space is local to the process. Thus, two
+    // processes do mmap with the same offset get different mappings.
+    mr->handle = info->next_offset;
```

Comment:
can we move this code to an helper function?
besides, there is a wrap-around problem in this design, isn't there?
it should be at least documented
lacking a better solution, you should return an error when wrapping